src/database/utils/migrationConfigLoader.ts

```
@@ -0,0 +1,84 @@
+   */
+  static async loadMigrationsFromDirectory(): Promise<Migration[]> {
+    try {
+      // This would require a server-side implementation or build-time processing
+      // For now, we'll use the JSON config approach as it's more practical for client-side
+      throw new Error('Directory-based migration loading is not implemented for client-side environments');
+    } catch (error) {
+      console.error('Failed to load migrations from directory:', error);
+      throw error;
+    }
+  }
+} 
```

Biome (1.9.4)

[error] Avoid classes that contain only static members.

Prefer using simple functions instead of classes with only static members.

(lint/complexity/noStaticOnlyClass)